### PR TITLE
WEBUI-971: distinguish view vs download in history tab

### PR DESCRIPTION
**Pull request by swarnadipa-dev**

### CODE REVIEW

**Review by rakeshkumar1019 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by satishyadav-dev — DISMISSED** [review]

**Review by Pranit-Sotre — COMMENTED** [review]
Please look into below issues:
1. Storyboard downloads (videos) are not getting  regrouped with the "[user] viewed the document" entry in activity tab.
2. Download entry is created when downloaded from main UI. User downloads another renditions then new entry for download is getting created in activity tab.

**Review by Pranit-Sotre — DISMISSED** [review]

File: elements/nuxeo-document-activity/nuxeo-document-activity.js [hunk 1]
```diff
@@ -105,6 +107,11 @@ Polymer({
   _areGatherableActivities(a, b) {
     let delta = new Date(a.eventDate) - new Date(b.eventDate);
     delta = delta / 1000 / 60 / 60; // Converts ms to hours
-    return a.eventId === b.eventId && a.eventId === 'download' && a.principalName === b.principalName && delta < 24;
+    return (
+      (a.extended.clientReason === 'view' || a.extended.clientReason === 'download') &&
+      a.extended.clientReason === b.extended.clientReason &&
```
Comment (rajatgoel-gl) [inline elements/nuxeo-document-activity/nuxeo-document-activity.js:116]: 'null' or 'undefiened' checks are missing

File: elements/nuxeo-audit/nuxeo-audit-search.js [hunk 2]
```diff
@@ -264,11 +266,13 @@ class AuditSearch extends mixinBehaviors([FormatBehavior, RoutingBehavior], Nuxe
     return null;
   }
 
-  _formatComment(comment) {
+  _formatComment(comment, clientReason) {
     if (moment(comment, moment.ISO_8601).isValid()) {
       return this.formatDateTime(comment);
     }
-    return comment;
+    return clientReason && clientReason.toLowerCase().includes('view')
```
Comment (rajatgoel-gl) [inline elements/nuxeo-audit/nuxeo-audit-search.js]: I assume we are not supporting Internet Explorer 11 (or earlier). Kindly cross check our supported browsers.

File: elements/document/nuxeo-picture-formats.js [hunk 3]
```diff
@@ -88,7 +88,7 @@ Polymer({
             dimensions: `${view.info.width} x ${view.info.height}`,
             size: this.formatSize(view.content.length),
             format: view.info.format,
-            data: view.content.data,
+            data: `${view.content.data}&clientReason=download`,
```
Comment (rajatgoel-gl) [inline elements/document/nuxeo-picture-formats.js]: Kindly replace the hard coded repetition with the Generic functionality

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

File: elements/nuxeo-audit/nuxeo-audit-search.js (hunk 2 above)
Comment (swarnadipa-dev) [reply elements/nuxeo-audit/nuxeo-audit-search.js]: We do not support IE11. Still I have replaced 'includes' with 'indexOf'

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Review by rajatgoel-gl — APPROVED** [review]

**Review by rakeshkumar1019 — APPROVED** [review]

**Review by GuruprasadAnk1996 — APPROVED** [review]

**Review by satishyadav-dev — APPROVED** [review]

**Review by poonamyadav252 — APPROVED** [review]

**Review by Nisha-QA — APPROVED** [review]

**Pull request merged**